tests/src/performance/Scenario/JitBench/JitBenchHarness.cs

```diff
@@ -0,0 +1,487 @@
+
+                    if (File.Exists(sourceFilePath))
+                    {
+                        File.Copy(sourceFilePath, targetFilePath, true);
+                        Console.WriteLine($"    Copied file - '{targetFilePath}'");
+                    }
+                }
+            }
+        }
+
+        private static IDictionary<string, string> GenerateStore(IDictionary<string, string> environment)
+        {
+            // This step generates some environment variables needed later.
+            var environmentFileName = "JitBenchEnvironment.txt";
+            var psi = new ProcessStartInfo() {
+                WorkingDirectory = s_jitBenchDevDirectory,
+                FileName = "powershell.exe",
+                Arguments = $"-Command \".\\AspNet-GenerateStore.ps1 -InstallDir .store -Architecture {s_targetArchitecture} -Runtime win7-{s_targetArchitecture}; gi env:JITBENCH_*, env:DOTNET_SHARED_STORE | %{{ \\\"$($_.Name)=$($_.Value)\\\" }} 1>>{environmentFileName}\""
+            };
+
+            LaunchProcess(psi, 900000, environment);
+
+            // Return the generated environment variables.
+            return GetEnvironment(environment, Path.Combine(s_jitBenchDevDirectory, environmentFileName));
+        }
+
+        private static IDictionary<string, string> GetEnvironment(IDictionary<string, string> environment, string fileName)
+        {
+            foreach (var line in File.ReadLines(fileName))
+            {
+                if (string.IsNullOrWhiteSpace(line))
+                    continue;
+
+                string[] pair = line.Split(new char[] { '=' });
+                if (pair.Length != 2)
+                    throw new InvalidOperationException($"AspNet-GenerateStore.ps1 did not generate the expected environment variable {pair}");
+
+                if (!environment.ContainsKey(pair[0]))
+                    environment.Add(pair[0], pair[1]);
+                else
+                    environment[pair[0]] = pair[1];
+            }
+
+            return environment;
+        }
+
+        private static void RestoreMusicStore(string workingDirectory, string dotnetFileName, IDictionary<string, string> environment)
+        {
+            var psi = new ProcessStartInfo() {
+                WorkingDirectory = workingDirectory,
+                FileName = dotnetFileName,
+                Arguments = "restore"
+            };
+
+            LaunchProcess(psi, 300000, environment);
+        }
+
+        private static void PublishMusicStore(string workingDirectory, string dotnetFileName, IDictionary<string, string> environment)
+        {
+            var psi = new ProcessStartInfo() {
+                WorkingDirectory = workingDirectory,
+                FileName = "cmd.exe",
+                Arguments = $"/C \"{dotnetFileName} publish -c Release -f netcoreapp2.0 --manifest %JITBENCH_ASPNET_MANIFEST% /p:MvcRazorCompileOnPublish=false\""
```

Comment:
We should hoist `netcoreapp2.0` as a constant that we may need to update at the same time as the checksum of the JitBench repo, as it might change moving forward when we update the test sources.